File: PHP/logicaCondicional/Switch.php
<!-- Quando você precisa comparar uma única variável contra muitos valores específicos, a instrução switch oferece uma alternativa mais limpa para longas cadeias de if-elseif.

A instrução switch verifica uma variável uma vez e depois a compara contra diferentes valores case: -->
<?php
$day = "Monday";

switch ($day) {
    case "Monday":
        echo "Start of the week";
        break;
    case "Friday":
        echo "Almost weekend";
        break;
    case "Saturday":
    case "Sunday":
        echo "Weekend!";
        break;
    default:
        echo "Regular day";
}
?>
<!-- O PHP compara $day contra cada case. Quando encontra uma correspondência, executa o código desse bloco. A instrução break é crucial — ela diz ao PHP para sair do switch após executar o case correspondente. Sem ela, o PHP continuaria executando todo o código abaixo, mesmo de outros cases.

Observe como sábado e domingo compartilham o mesmo código ao empilhar cases juntos. O case default age como um else — ele executa quando nenhum outro case corresponde. -->
<?php
$grade = "B";

switch ($grade) {
    case "A":
        echo "Excellent!";
        break;
    case "B":
        echo "Good job!";
        break;
    default:
        echo "Keep trying!";
}
?>
<!-- Use switch ao comparar uma variável contra múltiplos valores exatos. Para condições complexas ou intervalos, fique com instruções if-elseif. -->

<!--
Desafio

Fácil
Leia uma string command da entrada representando a cor de um semáforo.

Use uma instrução switch para determinar a ação apropriada com base na cor:

Se o comando for red, imprima Stop
Se o comando for yellow, imprima Slow down
Se o comando for green, imprima Go
Se o comando for flashing red ou flashing yellow, imprima Proceed with caution
Para qualquer outro valor, imprima Invalid signal
Exemplo:

Se a entrada for green, a saída deve ser:

Go
Se a entrada for flashing red, a saída deve ser:

Proceed with caution
Se a entrada for blue, a saída deve ser:

Invalid signal -->
<?php
// Read input
$command = trim(fgets(STDIN));

// TODO: Write your code below using a switch statement
// Determine the appropriate action based on the traffic light color
switch ($command) {
    case "red":
        echo "Stop";
        break;  
    case "yellow":
        echo "Slow down";
        break;
    case "green":
        echo "Go";
        break;
    case "flashing red":
    case "flashing yellow":
        echo "Proceed with caution";
        break;
    default:
        echo "Invalid signal";
}

?>

File: Dart/FluxodeControle-TomadadeDecisões/switch.dart
/*
A instrução switch no Dart oferece uma alternativa mais limpa a múltiplas instruções if-else quando se verifica uma variável contra diferentes valores.

void main() {
  String grade = 'B';
  
  switch (grade) {
    case 'A':
      print('Excellent performance!');
      break;
    case 'B':
      print('Good performance!');
      break;
    case 'C':
      print('Average performance.');
      break;
    case 'D':
      print('Below average performance.');
      break;
    default:
      print('Invalid grade.');
      break;
  }
}
Quando executado, produz: Good performance!


Desafio

Fácil
Crie um programa que determina a nota de um aluno com base na pontuação do exame usando uma switch instrução:

Declare uma variável inteira chamada examScore com um valor de 85
Crie uma variável string chamada grade para armazenar a letra da nota
Use uma switch instrução com as seguintes regras:
Se examScore estiver entre 90 e 100 (inclusivo), defina grade para "A"
Se examScore estiver entre 80 e 89 (inclusivo), defina grade para "B"
Se examScore estiver entre 70 e 79 (inclusivo), defina grade para "C"
Se examScore estiver entre 60 e 69 (inclusivo), defina grade para "D"
Para qualquer outra pontuação, defina grade para "F"
Imprima o resultado neste formato exato: Score: 85, Grade: B
Nota: Você precisará usar divisão inteira para converter a pontuação para um intervalo que funcione com switch.
*/
void main() {
  // Declare the examScore variable here
  int examScore = 85;
  
  // Declare the grade variable here
  String grade;
  
  // Use a switch statement to determine the grade
  // A divisão inteira (examScore ~/ 10) simplifica os intervalos
  switch (examScore ~/ 10) {
    case 10:
    case 9:
      grade = "A";
      break;
    case 8:
      grade = "B";
      break;
    case 7:
      grade = "C";
      break;
    case 6:
      grade = "D";
      break;
    default:
      grade = "F";
      break;
  }
  
  // Print the result
  print("Score: $examScore, Grade: $grade");
}
